tests/src/genesis.rs

```diff
@@ -0,0 +1,197 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use bytecheck::CheckBytes;
+use dusk_core::abi::{ContractError, StandardBufSerializer};
+use dusk_core::abi::{ContractId, CONTRACT_ID_BYTES};
+use dusk_core::signatures::bls::{
+    PublicKey as AccountPublicKey, SecretKey as AccountSecretKey,
+};
+use dusk_core::stake::STAKE_CONTRACT;
+use dusk_core::transfer::data::ContractCall;
+use dusk_core::transfer::moonlight::AccountData;
+use dusk_core::transfer::{Transaction, TRANSFER_CONTRACT};
+use dusk_core::{dusk, LUX};
+use dusk_vm::{execute, ExecutionConfig};
+use dusk_vm::{CallReceipt, ContractData, Error as VMError, Session, VM};
+use rkyv::validation::validators::DefaultValidator;
+use rkyv::{Archive, Deserialize, Infallible, Serialize};
+
+use crate::common::utils::{account, chain_id};
+
+const ZERO_ADDRESS: ContractId = ContractId::from_bytes([0; CONTRACT_ID_BYTES]);
+const GAS_LIMIT: u64 = 0x10000000;
+const CHAIN_ID: u8 = 0x1;
+const MOONLIGHT_GENESIS_VALUE: u64 = dusk(1_000.0);
+const MOONLIGHT_GENESIS_NONCE: u64 = 0;
+const NO_CONFIG: ExecutionConfig = ExecutionConfig::DEFAULT;
+
+type Result<T, Error = VMError> = core::result::Result<T, Error>;
+
+/// VM Sessions that behaves like a mainnet VM.
+///
+/// Calls go as tx through the genesis transfer contract,
+/// before reaching the non-genesis contract.
+pub struct MainnetSession {
```

Comment:
I wouldn't call it `MainnetSession` since it can also be used to test against testnet or devnet. Maybe `NetworkSession`?